src/core/net/srp_server.cpp

```diff
@@ -1196,13 +1197,23 @@ void Server::HandleUpdate(Host &aHost, const MessageMetadata &aMetadata)
     {
         UpdateMetadata *update = UpdateMetadata::Allocate(GetInstance(), aHost, aMetadata);
 
-        mOutstandingUpdates.Push(*update);
-        mOutstandingUpdatesTimer.FireAtIfEarlier(update->GetExpireTime());
+        if (update == nullptr)
+        {
+            error = kErrorNoBufs;
```

Comment:
nit:
```suggestion
            CommitSrpUpdate(kErrorNoBufs, aHost, aMetadata);
```

So that we can avoid the new `shouldCommit` variable (we are depending on the `shouldCommit`'s initial value (`true`) which is far away from here it is used... it's just a little bit readability issue

Reply:
@wgtdkp , I moved the code at `exit` into a method (so we can use `exit` again). I think this helps make the code simpler. PTAL.

Reply:
@wgtdkp , I moved the code at `exit` into a method (so we can use `exit` again). I think this helps make the code simpler. PTAL.

Reply:
LGTM, thanks!